code/game/machinery/gumball.dm

```diff
@@ -0,0 +1,115 @@
+/obj/machinery/gumballmachine
+	name = "gumball machine"
+	desc = "A colorful candy machine."
+	icon = 'icons/obj/vending.dmi'
+	icon_state = "gumball_100"
+	layer = 2.9
+	anchored = 1
+	density = 1
+	use_power = 1
+	idle_power_usage = 10
+	var/typeofcandy = "gumballs"
+	var/initialicon = "gumball"
+	var/amountleft = 20
+	var/vendingtype = /obj/item/clothing/mask/chewable/candy/gum
+	var/gumprice = 5
+	var/on = 1
+	var/broken = 0
+
+/obj/machinery/gumballmachine/Initialize()
+	. = ..()
+
+
+/obj/machinery/gumballmachine/machinery_process()
+	if(broken)
+		return
+	else
+
+		update_power()
+
+		update_icon()
+
+
+/obj/machinery/gumballmachine/examine()
+	..()
+	to_chat(usr, span("notice", "\The [src] costs [gumprice] credits to use."))
```

Comment:
Use of `usr` in a non-interfacing proc.

Psure examine passes a user argument, use that.